app/assets/javascripts/modules/inline-image-modal.js

```diff
@@ -63,6 +100,15 @@ InlineImageModal.prototype.performAction = function (item) {
   handlers[item.dataset.modalAction].bind(this)()
 }
 
+InlineImageModal.prototype.initComponents = function () {
+  // TODO: change govuk-frontend so we can do GOVUKFrontend.initAll(dynamicSection)
+  var $errorSummary = document.querySelector('[data-module="error-summary"]')
+  new window.GOVUKFrontend.ErrorSummary($errorSummary).init()
+
+  // TODO: change ErrorSummary to listen on the scoped element (not static window)
+  window.dispatchEvent(new window.Event('load'))
```

Comment:
Oh gosh - I see what you mean now about the `window.load` thing we were discussing yesterday. This is rather unusual to synthetically call the page load event as there is an anticipation it will only be called once.

Btw that couldn't be changed to be listen to a scope as I think only window has a load event, but it could have a separate function called to initialise.

Anyway from looking at [govuk-frontend/error-summary.js](https://github.com/alphagov/govuk-frontend/blob/master/src/components/error-summary/error-summary.js) I do wonder if it'll be doing some things that are either undesirable or problematic in a modal context. 

The undesirable aspect would be that it adds to the history state which enables back button access. I don't know fully how this works, however I'd anticipate that for us, we could end up with a back button that does nothing once the modal isn't displayed.

The other thing that I imagine either just works fine or is painful is the auto scrolling that the error summary does since the modal is in a position fixed I don't know if it'd scroll the window or the element we want.

Reply:
@kevindew agreed, it's not ideal. I reckon it's actually unnecessary to listen to the event, and the `focus()` can just be called immediately - @alex-ju reckons it's a relic from an older version of this component, and we're going to try and remove the event listener.

The scrolling seems to work fine, especially as we expect the modal to occupy most of the page anyway. I hadn't noticed the 'back' behaviour though, which isn't an issue for the inline image work, as none of the error summaries have any links in their content. I think this is something we could resolve (i.e. turn off) if the component was better isolated from the window, so I'm not too worried about it, although I'll be careful to check this in our local ones.

Reply:
I proposed a change in `govuk-frontend` to [remove the dependency on `document`](https://github.com/alphagov/govuk-frontend/pull/1215), but I get that that that's there to make sure the component is not being initialised twice.

Reply:
Rather than having this synthetic load event could we do `$errorSummary.focus()` ?

That feels like it'd less potential consequences and we're already coupling this load behaviour to that particular component.

Reply:
Done - turns out window events are only used by two components, and we're hoping to get rid of it on this one, so the intent is no longer to 'simulate a window loading'.